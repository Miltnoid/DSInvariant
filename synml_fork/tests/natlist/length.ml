type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

let length : list -> nat |>
  { [] => 0
  | [0] => 1
  | [0;0] => 2 } = ?
