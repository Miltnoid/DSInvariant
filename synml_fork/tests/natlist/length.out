(* Generated Wed Dec 31 11:16:14 2014 *)
let length : list -> nat =
  let rec f1 (l1:list) : nat =
    match l1 with
      | Nil -> O
      | Cons (n1, l2) -> S (f1 l2)
  in
    f1
;;
